src/f32/funcs.rs

```diff
@@ -349,3 +465,25 @@ fn test_scalar_sin_cos() {
     assert!(s.is_nan());
     assert!(c.is_nan());
 }
+
+#[test]
+#[cfg(vec4sse2)]
+fn test_sse2_m128_sin() {
+    use crate::Vec4;
+    use std::f32::consts::PI;
+
+    fn test_sse2_m128_sin_angle(a: f32) {
+        let v = Vec4::splat(a);
+        let v = unsafe { Vec4(sse2::m128_sin(v.0)) };
+        assert_approx_eq!(v.x(), a.sin(), 1e-6);
+    }
+
+    let mut a = -PI;
```

Comment:
In `test_scalar_sin_cos` I'm testing a much larger range, and accuracy did turn out to be a problem for large input values. I think it would be better to copy that test, unless this sin is only being used by slerp and the input it expected to always be in the [-PI, PI] range then it might be OK to not include the `m128_mod_angles` call and not test a larger range. However it should be clearly commented that input needs to be [-PI, PI] and asserts added to check that is the case. Otherwise that `m128_mod_angles` call probably needs to got back in and this test should test across a large range of values. I guess if this is all sounding like too much work you could just use `f32:sin` and leave an optimised approximation for another day :)